### ISSUE TITLE
nbest in rnn

### ISSUE THREAD

**Issue by versis** [fchollet/keras#1316]
Hi, I'm just starting to learn how keras is working and I have few questions.
1. How exactly is **preditct_prob()** working? Especially for recurrent layers.
   After training I would like to compare sequences and check which one is more probable given the model. Is this method providing this funcionality? I can't figure this out from the code :/
2. Another question regarding example [**lstm_text_generation.py**](https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py).  How many layers this network contains? I see 2xLSTM and 1xDense, but do we also asume that there is input layer and output layer?
   _EDITED:_
   I see that the first layer always specify the input so it does the job for two layers. I guess output of the Dense layer here is output of the whole network, then in summary we have 4 layers (input, LSTM, LSTM, output). Am I right?
3. Second LSTM (in the same example) looks like this:
   _model.add(LSTM(512, return_sequences=False))_
   Why **return_sequence** is **False**? Don't we want the network to backpropagate for every element in sequence in each layer?
4. Is there any parameter for truncated backpropagation?

**Comment by versis** [comment]
_**Update for point 3:**_
I noticed that there is a possibility to have either **Dense** or **TimeDistributedDense** layer at the end of the network. In the first case we would like to have return_sequence equal to _False_ and in the second case _True_, correct?

Do we assume that when using TimeDistributedDense labels (y) are sequences of the same size as corresponding input sequences (x)?

If so, then using Dense at the end (given that I want to have RNN) means that we feed the network with sequence and expect one element at the end, correct? 

Putting this in context of [http://karpathy.github.io/2015/05/21/rnn-effectiveness/](Karpathy's rrn article) and his description of different types of RNN. Using Dense is like _many to one (3)_ type, and using TimeDistributedDense is like _many to many (5, not 4)_ type?

Why isn't the example from point **2** using TimeDistributedDense layer at the end? Wouldn't that work better (in theory)?
